README.md

```diff
@@ -1,130 +1,182 @@
+
+To make requests to a vehicle from a web application the end user must connect their vehicle using [Smartcar's authorization flow](https://smartcar.com/docs#authentication). The Smartcar Javascript SDK provides two scripts that assist with running Smartcar's authorization flow.
+
+- `sdk.js`: Sourced on the page your authorization flow begins from. Attaches to the browser's window as `Smartcar`. Allows you to generate a Smartcar OAuth URL and launch Smartcar's authorization flow in a pop-up.
+- `redirect.js`: Sourced on redirect page following completion of OAuth flow. Runs on load. This *optional* file can be served in the last page of your OAuth flow to close the popup window and trigger firing of an optional `onComplete` method passed to the `Smartcar` constructor from `sdk.js`. If you use a Smartcar hosted redirect (explained in greater detail below) this file is automatically served and the `onComplete` method becomes required.
 
 For more information about Smartcar's authorization flow, please visit our [API documentation](https://smartcar.com/docs#authentication).
 
 ## Quick Start
-Before integrating with Smartcar's SDK, you'll need to register an application in the [Smartcar Developer portal](https://developer.smartcar.com). If you do not have access to the dashboard, please [request access](https://smartcar.com/subscribe).
 
-The SDK helps ease the [OAuth authorization process](https://tools.ietf.org/html/rfc6749#section-4.1). A sample flow looks like this:
+Before integrating with Smartcar's SDK, you'll need to register an application in the [Smartcar Developer portal](https://dashboard.smartcar.com).
+
+The SDK helps ease the [OAuth authorization process](https://tools.ietf.org/html/rfc6749#section-4.1).
+
+We offer a choice of using a Smartcar hosted redirect or hosting your own. This choice is made when you add redirect URIs in the developer dashboard and influences how you use this SDK. As a developer there are several key differences between the two strategies.
+
+First, if you use Smartcar hosting you will add a redirect URI to your application of the form `https://javascript-sdk.smartcar.com/redirect?app_origin=<your-client-uri-here>`. So for example, if your application URI was `https://my-awesome-app.com` you'd add the following redirect URI to your app in the dashboard: `https://javascript-sdk.smartcar.com/redirect?app_origin=https://my-awesome-app.com`. Note, this value is only your app's origin, if your application was at `https://my-awesome-app.com/application` you would still have the example redirect URI listed without the `/application`. Any restrictions on normal redirect URIs (like requiring `https` when not using `localhost` still apply to the value you give for `app_origin`). See the [Register](https://smartcar.com/docs#register) section of our docs for valid formats.
```

Comment:
Outside of explaining _how_ to use Smartcar hosting, do we want to explain _what_ it is for someone who doesn't know. Like what the fundamental differences are and why a developer would choose one over another.